corehq/blobs/mixin.py

```diff
@@ -447,3 +449,9 @@ def bulk_atomic_blobs(docs):
 @memoized
 def _get_couchdb_name(doc_class):
     return doc_class.get_db().dbname
+
+
+def safe_id(identifier):
+    if not SAFENAME.match(identifier):
+        identifier = u'sha1-' + sha1(identifier.encode('utf-8')).hexdigest()
```

Comment:
could we fail hard if a natural doc id starts with `sha1-`? that probably never happens and would safeguard against blowing away data